app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddPathSurfaceForm.kt

```diff
@@ -1,56 +1,43 @@
 import de.westnordost.streetcomplete.view.image_select.Item
-import de.westnordost.streetcomplete.quests.surface.Surface.*
+import de.westnordost.streetcomplete.util.TextChangedWatcher
 
-class AddPathSurfaceForm : AGroupedImageListQuestAnswerFragment<String, String>() {
+class AddPathSurfaceForm : AImageListQuestAnswerFragment<String, DetailSurfaceAnswer>() {
+    override val items: List<Item<String>> get() =
+        (PAVED_SURFACES + UNPAVED_SURFACES + GROUND_SURFACES).toItems() +
+            Item("paved", R.drawable.path_surface_paved, R.string.quest_surface_value_paved, null, listOf()) +
+            Item("unpaved", R.drawable.path_surface_unpaved, R.string.quest_surface_value_unpaved, null, listOf()) +
+            Item("ground", R.drawable.surface_ground, R.string.quest_surface_value_ground, null, listOf())
 
-    override val topItems get() =
-        when (val pathType = determinePathType(osmElement!!.tags)) {
-            "bridleway" -> listOf(
-                DIRT, GRASS, SAND,
-                PEBBLES, FINE_GRAVEL, COMPACTED
-            )
-            "path" -> listOf(
-                DIRT, PEBBLES, COMPACTED,
-                ASPHALT, FINE_GRAVEL, PAVING_STONES
-            )
-            "footway" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                COMPACTED, FINE_GRAVEL, DIRT
-            )
-            "cycleway" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                COMPACTED, WOOD, METAL
-            )
-            "steps" -> listOf(
-                PAVING_STONES, ASPHALT, CONCRETE,
-                WOOD, SETT, UNHEWN_COBBLESTONE
-            )
-            else -> throw IllegalStateException("Unexpected path type $pathType")
-        }.toItems()
+    override val itemsPerRow = 3
 
-    override val allItems = listOf(
-        // except for different panorama images, should be the same as for the road quest, to avoid confusion
-        Item("paved", R.drawable.panorama_path_surface_paved, R.string.quest_surface_value_paved, null, PAVED_SURFACES.toItems()),
-        Item("unpaved", R.drawable.panorama_path_surface_unpaved, R.string.quest_surface_value_unpaved, null, UNPAVED_SURFACES.toItems()),
-        Item("ground", R.drawable.panorama_surface_ground, R.string.quest_surface_value_ground, null, GROUND_SURFACES.toItems())
-    )
-
-    private fun determinePathType(tags: Map<String, String>): String? {
-        val pathType = tags["highway"]
-        // interpret paths with foot/bicycle/horse=designated as...
-        if ("path" == pathType) {
-            if ("designated" == tags["bicycle"]) return "cycleway"
-            if ("designated" == tags["horse"]) return "bridleway"
-            if ("designated" == tags["foot"]) return "footway"
+    override fun onClickOk(selectedItems: List<String>) {
+        val value = selectedItems.single()
+        if(value == "paved" || value == "unpaved" || value == "ground") {
```

Comment:
`ground` as well? But in the element filter, you don't search for `ground`.

Reply:
So this means that you should remove `|| value == "ground"` here and in the other places where you added it

Reply:
And find new image for `ground` or drop it as an answer (if treated as synonym of `dirt`)

Reply:
I re-read the wiki for [`surface`](https://wiki.openstreetmap.org/wiki/Key:surface), and I think `ground` should be dropped. The description (at least in English) basically asks you to use a different value.

Reply:
Thanks for pointing me toward wiki statement about `ground` value.

Wiki matches rather what was originally implemented by @westnordost rather what I think.

I will start treating it as a generic value, like `surface=paved` and `surface=unpaved`.

Reply:
I have no strong preference (for now at least, I may change my mind after using the new form of a quest for some time), so I would implement here whatever version you prefer.

But I am not entirely sure what is the preferred solution:

Reasking for `surface=ground` is unwanted.

Asking for detailed note for `surface=ground` is unwanted.

Is it necessary to keep "Are you sure that you can not determine it more specifically?" for "Ground (generic)" or some equivalent?

Is it necessary to find a new image for `ground` answer?

Reply:
> write something like "mix of leaves, roots, divots"

I humbly disagree. In my opinion, this is `dirt` with an appropriate `smoothness` tag.

> it will differ all the time. A few meters grass, then more earth, then some pebbles and rocks, then a puddle of mud

This is a reasonable point. But now I am in the same place as @matkoniecz. See his questions above.

Reply:
For now old image is used, there is no confirmation on selecting `ground`, resurvey for `ground` is not done.

Reply:
I think this is fine